fireapp/lib/presentation/constraint_form/constraint_form_view_model.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../../domain/request_state.dart';
+
+class ConstraintFormViewModel {
+  final GlobalKey<FormState> formKey = GlobalKey<FormState>();
+
+  // Text Editing Controllers
+  final TextEditingController titleController = TextEditingController();
+  final TextEditingController inputDateController = TextEditingController();
+  final TextEditingController startTimeController = TextEditingController();
+
+  // RequestState stream
+  final BehaviorSubject<RequestState<void>> _state = BehaviorSubject.seeded(RequestState.initial());
+  Stream<RequestState<void>> get state => _state.stream;
+
+  // Dummy values for the assets list
+  final Stream<List<int>> assetsStream = Stream<List<int>>.value([1, 2, 3, 4]);
```

Comment:
This should be a list of Asset objects (which I forgot to ask you to write). I'll make a PR into your branch with a model, then please update this.

Reply:
PR #48 contains this change